iceoryx_dust/source/posix_wrapper/message_queue.cpp

```diff
@@ -24,77 +25,83 @@
-                           const IpcChannelSide channelSide,
-                           /// NOLINTJUSTIFICATION @todo iox-#832 should be solved when the arguments are put in a
-                           ///                      struct
-                           /// NOLINTNEXTLINE(bugprone-easily-swappable-parameters)
-                           const size_t maxMsgSize,
-                           const uint64_t maxMsgNumber) noexcept
-    : m_channelSide(channelSide)
+expected<MessageQueue, IpcChannelError> MessageQueueBuilder::create() const noexcept
 {
-    sanitizeIpcChannelName(name)
-        .and_then([this](IpcChannelName_t& name) { this->m_name = std::move(name); })
-        .or_else([this](IpcChannelError) {
-            this->m_isInitialized = false;
-            this->m_errorValue = IpcChannelError::INVALID_CHANNEL_NAME;
-        });
+    auto sanitzedNameResult = MessageQueue::sanitizeIpcChannelName(m_name);
+    if (sanitzedNameResult.has_error())
+    {
+        return err(IpcChannelError::INVALID_CHANNEL_NAME);
+    }
+    auto& sanitizedName = sanitzedNameResult.value();
+    IOX_MAYBE_UNUSED std::false_type m_name; // m_name shall not be used anymore but only sanitizedName
 
-    if (maxMsgSize > MAX_MESSAGE_SIZE)
+    if (m_maxMsgSize > MessageQueue::MAX_MESSAGE_SIZE)
     {
-        this->m_isInitialized = false;
-        this->m_errorValue = IpcChannelError::MAX_MESSAGE_SIZE_EXCEEDED;
+        return err(IpcChannelError::MAX_MESSAGE_SIZE_EXCEEDED);
     }
-    else
+
+    if (m_channelSide == IpcChannelSide::SERVER)
     {
-        if (channelSide == IpcChannelSide::SERVER)
-        {
-            posixCall(mq_unlink)(m_name.c_str())
-                .failureReturnValue(ERROR_CODE)
-                .ignoreErrnos(ENOENT)
-                .evaluate()
-                .and_then([this](auto& r) {
-                    if (r.errnum != ENOENT)
-                    {
-                        std::cout << "MQ still there, doing an unlink of " << m_name << std::endl;
-                    }
-                });
-        }
-        // fields have a different order in QNX,
-        // so we need to initialize by name
-        m_attributes.mq_flags = 0;
-        m_attributes.mq_maxmsg = static_cast<long>(maxMsgNumber);
-        m_attributes.mq_msgsize = static_cast<long>(maxMsgSize);
-        m_attributes.mq_curmsgs = 0L;
+        posixCall(mq_unlink)(sanitizedName.c_str())
+            .failureReturnValue(MessageQueue::ERROR_CODE)
+            .ignoreErrnos(ENOENT)
+            .evaluate()
+            .and_then([&sanitizedName](auto& r) {
+                if (r.errnum != ENOENT)
+                {
+                    std::cout << "MQ still there, doing an unlink of " << sanitizedName << std::endl;
+                }
+            });
+    }
+
+    // fields have a different order in QNX, so we need to initialize by name
+    mq_attr attributes;
+    attributes.mq_flags = 0;
+    attributes.mq_maxmsg = static_cast<long>(m_maxMsgNumber);
```

Comment:
From the point of platform indepence wouldn't it be better to use `static_cast<decltype(attributes.mq_maxmsg)>(m_maxMsgNumber)`. The reason is that another platform maybe uses here `unsigned long` or `int` and then we introduce a warning on that platform.

Some goes for that statement below.